src/api/middlewares/middlewares.auth.js

```diff
@@ -55,3 +55,83 @@ export const checkIfEmailAlreadyExist = async (req, res, next) => {
     logger.error(`checking existing email failed::${enums.CHECK_EXISTING_EMAIL}`, error.message);
   }
 };
+
+export const emailDoesNotExist = async (req, res, next) => {
+  try {
+    const { email } = req.body;
+    const user = await authServices.findEmail([ email ]);
+    logger.info(`${enums.CURRENT_TIME_STAMP}, :::Info: user with email found`);
+
+    if (!user) return ApiResponse.error(res, enums.EMAIL_DOES_NOT_EXIST, enums.HTTP_BAD_REQUEST);
+
+    req.user = user;
+    return next();
+  } catch (error) {
+    error.label = enums.EMAIL_DOES_NOT_EXIST; 
+    logger.error(`checking if email exists::${enums.EMAIL_DOES_NOT_EXIST}`, error.message);
+  }
+};
+
+export const checkIfEmailVerified = async (req, res, next) => {
+  try {
+    const { user } = req;
+
+    if (!user.is_email_verified) 
+      return ApiResponse.error(res, enums.UNVERIFIED_EMAIL, enums.HTTP_BAD_REQUEST);
```

Comment:
there should be a logger showing that the email was verified successfully before returning your response